google-ads-examples/src/main/java/com/google/ads/googleads/examples/billing/AddBillingSetup.java

```diff
@@ -0,0 +1,263 @@
+      }
+    } catch (Exception e) {
+      System.err.printf("  Error: %s%n", e);
+    }
+  }
+
+  /**
+   * Runs the example.
+   *
+   * @param googleAdsClient the Google Ads API client.
+   * @param customerId the customer ID containing the BillingSetup to remove.
+   * @param paymentsAccountId optional payments account ID to attach to the new billing setup. Must
+   *     be formatted as "1234-5678-9012-3456".
+   * @param paymentsProfileId optional payments profile ID to attach to a new payments account and
+   *     to the new billing setup. Must be formatted as "1234-5678-9012".
+   */
+  private void runExample(
+      GoogleAdsClient googleAdsClient,
+      long customerId,
+      String paymentsAccountId,
+      String paymentsProfileId)
+      throws Exception {
+    // Constructs a new billing setup.
+    BillingSetup billingSetup =
+        createBillingSetup(customerId, paymentsAccountId, paymentsProfileId);
+
+    billingSetup = setBillingSetupDateTimes(googleAdsClient, customerId, billingSetup);
+
+    // Creates the billing setup operation.
+    BillingSetupOperation operation =
+        BillingSetupOperation.newBuilder().setCreate(billingSetup).build();
+
+    // Creates the service client.
+    try (BillingSetupServiceClient billingSetupServiceClient =
+        googleAdsClient.getLatestVersion().createBillingSetupServiceClient()) {
+      // Issues the mutate request to add the billing setup.
+      MutateBillingSetupResponse billingResponse =
+          billingSetupServiceClient.mutateBillingSetup(Long.toString(customerId), operation);
+
+      System.out.printf(
+          "Added new billing setup with resource name '%s'.%n",
+          billingResponse.getResult().getResourceName());
+    }
+  }
+
+  /**
+   * Creates and returns a new Billing Setup instance with complete payment details. One of
+   * paymentsAccountId or paymentsProfileId must be provided.
+   *
+   * @param customerId the customer ID containing the BillingSetup to remove.
+   * @param paymentsAccountId optional payments account ID to attach to the new billing setup. Must
+   *     be formatted as "1234-5678-9012-3456".
+   * @param paymentsProfileId optional payments profile ID to attach to a new payments account and
+   *     to the new billing setup. Must be formatted as "1234-5678-9012".
+   * @return new Billing Setup instance with complete payment details.
+   */
+  private BillingSetup createBillingSetup(
+      long customerId, String paymentsAccountId, String paymentsProfileId) throws Exception {
+    BillingSetup.Builder billingSetupBuilder = BillingSetup.newBuilder();
+
+    // Sets the appropriate payments account field.
+    if (paymentsAccountId != null) {
+      // If a payments account id has been provided, set PaymentsAccount to its resource
+      // name. You can list available payments accounts via the PaymentsAccountService's
+      // ListPaymentsAccounts method.
+      billingSetupBuilder.setPaymentsAccount(
```

Comment:
Has this been tested?

Reply:
I do not have access to an account with monthly invoicing, so I relied on the golden for this one. @AndrewMBurke can you verify this has been tested for DotNet?

@nwbirnie do you have a specific concern here?